src/org/labkey/test/components/glassLibrary/ParentEntityEditPanel.java

```diff
@@ -0,0 +1,475 @@
+        // div.Select-value,div.Select-input). That is if there is only one element selected. If it is a multi-select
+        // control each selected value will have div.Select-value.
+
+        List<ReactSelect> allControls = ReactSelect.finder(getDriver()).findAll(this);
+        List<ReactSelect> typeCombos = new ArrayList<>();
+        for(ReactSelect reactSelect : allControls)
+        {
+            try
+            {
+                if (reactSelect.getComponentElement().getAttribute("class").contains("Select--single"))
+                    typeCombos.add(reactSelect);
+            }
+            catch (NoSuchElementException | StaleElementReferenceException ex)
+            {
+                // Finder returned a react select that is stale or not in scope. Ignore this error.
+            }
+        }
+        return typeCombos;
+    }
+
+    /**
+     * Get the combo that will list the individual elements, either samples or sources, that will become a parent.
+     * This combo will only show up after a type is selected.
+     *
+     * @return A reference to the appropriate combo that will be for the new parent sample or source.
+     */
+    protected FilteringReactSelect getIdCombo()
+    {
+        // Will get the ID Combo for the last field. A new parent is added to the end.
+        int numOfTypes = numberOfTypeFields();
+
+        return FilteringReactSelect.finder(getDriver())
+                .followingLabelWithSpan(_parentType.getType() + " IDs")
+                .findAll(this).get(numOfTypes - 1);
+
+    }
+
+    /**
+     * Get the id combos by it's index (zero based). This would include any 'add new' combos. The parent id combo
+     * boxes do not have unique labels so there is no get based on the label.
+     *
+     * @return A combo at the given position in the colection of combos.
+     */
+    protected FilteringReactSelect getIdCombo(int index)
+    {
+        return getAllIdCombo().get(index);
+    }
+
+    //TODO: Should add a function to get the ID combo associated with a type name.
+
+    /**
+     * Get all of the id combos in the panel. This includes those already set and any 'add new' combos as well.
+     *
+     * @return A collection of all of the combos that select the sample or source ids.
+     */
+    protected List<FilteringReactSelect> getAllIdCombo()
+    {
+        // Calling ReactSelect.finder(getDriver()).findAll(this) is not specific enough, it returns all divs with
+        // 'select' in the class. That can return multiple controls for one react select (div.Select, div.Select-control,
+        // div.Select-value,div.Select-input). That is if there is only one element selected. If it is a multi-select
+        // control each selected value will have div.Select-value.
+
+        List<FilteringReactSelect> allControls = FilteringReactSelect.finder(getDriver()).findAll(this);
```

Comment:
Might be able to use your own Locator to avoid sifting through to find what you want.
```
List<ReactSelect> idCombos = FilteringReactSelect.finder(getDriver())
        .locatedBy(Locator.tag("div").withClasses("Select", "Select--multi"))
        .findAll(this);
```

Reply:
Good call! I changed the function to return this collection.